cradle-cassandra/src/main/java/com/exactpro/cradle/cassandra/dao/intervals/CassandraIntervalsWorker.java

```diff
@@ -0,0 +1,162 @@
+    private final IntervalOperator intervalOperator;
+
+    public CassandraIntervalsWorker(CassandraSemaphore semaphore, UUID instanceUuid, Function<BoundStatementBuilder,
+            BoundStatementBuilder> writeAttrs, Function<BoundStatementBuilder,
+            BoundStatementBuilder> readAttrs, IntervalOperator intervalOperator)
+    {
+        this.semaphore = semaphore;
+        this.instanceUuid = instanceUuid;
+        this.writeAttrs = writeAttrs;
+        this.readAttrs = readAttrs;
+        this.intervalOperator = intervalOperator;
+    }
+
+    @Override
+    public void storeInterval(Interval interval) throws IOException
+    {
+        try
+        {
+            storeIntervalAsync(interval).get();
+        }
+        catch (Exception e)
+        {
+            throw new IOException("Error while storing interval "+interval.getId(), e);
+        }
+    }
+
+    @Override
+    public CompletableFuture<Void> storeIntervalAsync(Interval interval)
+    {
+        CompletableFuture<IntervalEntity> future = new AsyncOperator<IntervalEntity>(semaphore)
+                .getFuture(() -> {
+                    IntervalEntity intervalEntity = new IntervalEntity(interval, instanceUuid);
+                    return intervalOperator.writeInterval(intervalEntity, writeAttrs);
+                });
+
+        return future.thenAccept(r -> {});
+    }
+
+    @Override
+    public Iterable<Interval> getIntervals(Instant from, Instant to, String crawlerName, String crawlerVersion) throws IOException
+    {
+        try
+        {
+            return getIntervalsAsync(from, to, crawlerName, crawlerVersion).get();
+        }
+        catch (Exception e)
+        {
+            throw new IOException("Error while getting healing intervals from " + from, e);
+        }
+    }
+
+    @Override
+    public CompletableFuture<Iterable<Interval>> getIntervalsAsync(Instant from, Instant to, String crawlerName, String crawlerVersion)
+    {
+        LocalDateTime fromDateTime = LocalDateTime.ofInstant(from, TIMEZONE_OFFSET),
+                toDateTime = LocalDateTime.ofInstant(to, TIMEZONE_OFFSET);
+
+        LocalTime fromTime = fromDateTime.toLocalTime(),
+                toTime = toDateTime.toLocalTime();
+
+        LocalDate date = fromDateTime.toLocalDate();
+
+        CompletableFuture<MappedAsyncPagingIterable<IntervalEntity>> future =
+                new AsyncOperator<MappedAsyncPagingIterable<IntervalEntity>>(semaphore)
+                        .getFuture(() -> intervalOperator
+                                .getIntervals(instanceUuid, date, fromTime, toTime, crawlerName, crawlerVersion, readAttrs));
+
+        return future.thenApply(entities -> {
+            try
+            {
+                return new IntervalsIteratorAdapter(entities);
+            }
+            catch (Exception error)
+            {
+                throw new CompletionException("Could not get healing intervals from "+from, error);
```

Comment:
They are not healing intervals, they are just intervals.
And you're getting them "from" and "to". `crawlerName` would also be useful in the error.